apps/explorer/components/FilterMobile.vue

```diff
@@ -74,8 +49,85 @@
 </template>
 <script lang="ts" setup>
 import Datepicker from '@vuepic/vue-datepicker'
+import { Soul } from '~~/types/soul'
+import { FilterParameters } from '~~/types/filter-parameters'
 
+const router = useRouter()
+const emit = defineEmits(['searchFilter'])
+const souls = $ref(useSouls())
 const createdDate = ref([])
 const updatedDate = ref([])
+let filteredResults = $ref([])
+const filterParams = $ref({} as FilterParameters)
+
+function filterResults() {
+  filteredResults = souls
+
+  router.push({
+    query: {
+      idNameOwner: filterParams.idNameOwner,
+      status: filterParams.status,
+      createdDate: filterParams.createdDate,
+      updatedDate: filterParams.updatedDate
+    }
+  })
+
+  if (filterParams.idNameOwner) {
+    filteredResults = filteredResults.filter(
+      (soul) =>
+        soul.id.toLocaleLowerCase().includes(filterParams.idNameOwner) ||
+        soul.name.toLocaleLowerCase().includes(filterParams.idNameOwner) ||
+        soul.owner.toLocaleLowerCase().includes(filterParams.idNameOwner)
+    )
+  }
+
+  if (filterParams.status) {
+    filteredResults = filteredResults.filter(
+      (soul) => soul.status === +filterParams.status
+    )
+  }
+
+  if (filterParams.createdDate) {
+    filteredResults = filteredResults.filter((soul) =>
+      soul.createdTimestamp.toString().includes(filterParams.createdDate)
+    )
+  }
+
+  if (filterParams.updatedDate) {
+    filteredResults = filteredResults.filter((soul) =>
+      soul.updatedTimestamp.toString().includes(filterParams.updatedDate)
+    )
+  }
+
+  emit('searchFilter', filteredResults as Soul[])
+}
+
+function searchResultByIdOrOwnerOrName(event): void {
+  const searchTerm = event.target.value
+  filterParams.idNameOwner = searchTerm
+
+  filterResults()
+}
+
+function searchResultByStatus(event): void {
+  const searchTerm = event.target.value
+  filterParams.status = searchTerm
+
+  filterResults()
+}
+
+function searchResultByCreatedDate(event): void {
+  const searchTerm = event.target.value
+  filterParams.createdDate = searchTerm
+
+  filterResults()
+}
+
+function searchResultByUpdatedDate(event): void {
+  const searchTerm = event.target.value
+  filterParams.updatedDate = searchTerm
+
+  filterResults()
+}
```

Comment:
Code duplication should be removed, isolate search functions into a separate file/package.